gazelle_cabal/dependency_resolution.go

```diff
@@ -224,42 +223,85 @@ func getPackageLabel(
 	pkgName string,
 	from label.Label,
 ) label.Label {
-	parentPackage, packageName := resolvePackageName(pkgName)
+	var resolvers []func(*resolve.RuleIndex, string, string, label.Label)(label.Label, error)
+	resolvers = append(resolvers,
+		resolveAgainstPrivateLibraries,
+		resolveAgainstPublicLibraries,
+		resolveAgainstRepository,
+	)
+
+	for _, resolver := range resolvers {
+		lbl, err := resolver(ix, packageRepo, pkgName, from)
+		if (err == nil) {
+			return lbl
+		}
+	}
+
+	log.Fatalf("Could not resolve dependency '%s' from '%s'", pkgName, from)
+	return label.Label{}
+}
+
+func resolveAgainstPrivateLibraries(
+	ix *resolve.RuleIndex,
+	packageRepo string,
+	pkgName string,
+	from label.Label,
+) (label.Label, error) {
+	parentPackage, depName := splitDependencyString(pkgName)
+
+	if len(parentPackage) > 0 && (parentPackage != depName) {
+		return label.Label{}, fmt.Errorf("It looks like %s is not an internal, private library", pkgName)
+	}
 
 	// Search for the rule of an internal library using the prefix "internal_library:" for the key
-	spec := resolve.ImportSpec{gazelleCabalName, "internal_library:" + packageName}
+	spec := resolve.ImportSpec{gazelleCabalName, "private_library:" + depName}
 	res := ix.FindRulesByImport(spec, gazelleCabalName)
 
 	// Search for the label of an internal library in the current package
 	for _, r := range res {
 		if r.IsSelfImport(from) {
-			// Cabal produces an error for circular dependency
 			log.Fatalf("Dependency cycle detected in the following component: %s", from)
 		}
 
 		// indeed internal library can be refereced form the package of its definition
-		// or it can be referenced using experimental colon syntax (a.k.a public internal libraries)
-		if (r.Label.Repo == from.Repo && r.Label.Pkg == from.Pkg) || (len(parentPackage) > 0) {
-			return rel(r.Label, from)
+		if (r.Label.Repo == from.Repo && r.Label.Pkg == from.Pkg) {
```

Comment:
The comment in line 266 needs editing.

Reply:
This line (and `if` statement) was removed in the latest commit.